### PR TITLE
Fix string hanlding for TCP Handler

### PR DESCRIPTION
**Pull request by sytone**
When using the connect command the comparisons have byte but the TcpHandler uses string which causes further breaks down the line. 

Update the Connect command and the TCPHandler to handle strings always and then the actual protocol will do the replacement as string and convert to byte for the banner part of the message.

### CODE REVIEW

**Comment by sytone** [comment]
I signed it!

File: adb/adb_protocol.py [hunk 1]
```diff
@@ -277,9 +277,10 @@ def Connect(cls, usb, banner=b'notadb', rsa_keys=None, auth_timeout_ms=100):
       InvalidResponseError: When the device does authentication in an
           unexpected way.
     """
+    hostbanner = 'host::%s' % banner
```
Comment (fahhem) [inline adb/adb_protocol.py]: why leave out the null byte?

Comment (sytone) [inline adb/adb_protocol.py]: I'm trying to fix remote debug. The URL handler expects a string. The and command expects bytes and the protocol expects bytes. I was playing with different options.

**Comment by sytone** [comment]
Reverted changes and took different approach. Now all paths expect serial to be a byte array and also the hostname is now returned as byte array so further banner processing does not break.

**Comment by fahhem** [comment]
Thanks! Hopefully this also benefits USB string handling since it's common code and I can't reproduce those problems.

**Pull request merged**